### ISSUE TITLE
classification optimizer (F1, ROCAUC, Accuracy, PR AUC)

### ISSUE THREAD

**Issue by josaum** [jquant/jai-sdk#261]
**Is your feature request related to a problem? Please describe.**
After running j.setup, with either db_type='classification' or db_type='metric_classification', the user should be able to:

- Generate Curves and Optimal Classification Thresholds for each Metric

Accuracy - from sklearn.metrics import confusion_matrix, accuracy_score
![image](https://user-images.githubusercontent.com/8642345/128510132-2d5a9f9b-2d27-4243-a852-2f99bc23abef.png)

F1 - from sklearn.metrics import f1_score
![image](https://user-images.githubusercontent.com/8642345/128510336-c00b7c38-53e9-4486-9d75-38a45e205f5b.png)

ROCAUC - from sklearn.metrics import roc_auc_score
![image](https://user-images.githubusercontent.com/8642345/128510452-dda0a9bd-0600-4e59-81d8-9301c87dcbd8.png)

PR AUC - from sklearn.metrics import average_precision_score
![image](https://user-images.githubusercontent.com/8642345/128510491-de09bb1c-397d-4d32-9b17-c18cf661aedb.png)

- Get inference (j.predict) already optimized for one of the options above.

optimized_for = [None, f1, rocauc, acc, prauc]

Ex: j.predict(name='collection', data=data, as_frame=True, predict_proba=True, optimized_for = 'f1')

**Describe the solution you'd like**

A reference implementation of the optimizer (by @ricardokazu):

`import matplotlib.pyplot as plt
import numpy as np

class Optimizer(object):
    def __init__(self, metric, n_iter=1001, maximize=True):
        self.metric = metric
        self.n_iter = n_iter
        self.thresholds = []
        self.metrics = []
        self.maximize = maximize

    def optimize(self, y_pred, y_test):
        for _th in np.linspace(0, 1, self.n_iter):
            _score = self.metric(y_test, (y_pred >= _th).astype(int))
            self.thresholds.append(_th)
            self.metrics.append(_score)

        if self.maximize:
            self.best_th = self.thresholds[np.argmax(self.metrics)]
        else:
            self.best_th = self.thresholds[np.argmin(self.metrics)]

    def plot(self, ax=None):
        if ax is None:
            ax = plt.axes()

        ax.axvline(self.best_th, color="r")
        ax.plot(self.thresholds, self.metrics, label="metric")
        ax.set_title("Metric per threshold")
        ax.legend()
        ax.set_ylabel(f"{self.metric.__class__.__name__}")
        ax.set_xlabel("thresholds")`

and usage:

`from sklearn.metrics import f1_score

predict_proba = model.predict_proba(...)

opt = Optimizer(f1_score)
opt.optimize(predict_proba[:, 1], y_test)
best_th = opt.best_th
opt.plot()`
